Picture frame step1

shim506 opened the pull request:
미션 1~4 구현 완료 리드미 첨부

xml 에서 id 작명과 코틀린 파일에서의 컴포넌트 작명이 어렵네요.
안드로이드 변수명에 대해서 참고할 만한 자료가 있을까요??

shim506 committed:
test

shim506 committed:
Logcat 활용 완료

shim506 committed:
TextView 구현하기 완료

shim506 committed:
Button 구현하기 완료

shim506 committed:
다른 Activity 로 이동 하기 완료

shim506 committed:
theme / style 학습 후 다크모드 구현

shim506 committed:
스택바 호출 위치 수정

ivybae commented:
@shim506 naming convention은 팀마다 정하기 나름이라 일반적으로 공개되는 자료는 없는것 같아요. 이 부분은 클래스 시간에 함께 이야기해볼게요~

ivybae commented on app/src/main/java/com/codesquad/kotlinphotoframe/MainActivity.kt:
```diff
@@ -1,11 +1,118 @@
+                AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
+            }
+        }
+    }
```
@shim506 
1. onCreate 함수 body에 newline이 너무 많은데요. 여러개의 빈줄을 사용할 수는 있지만 권장하지 않는 방식입니다. 수정해주세요~
2. onCreate 함수 body가 너무 길어져서 실행하는 코드 흐름이 잘 안보이네요. 이를 개선할 수 있는 방법이 있을까요?

shim506 replied:
- 필요하지 않은 빈줄을 정리했습니다.
- 코틀린 문법을 이용해서 if 문을 정리했습니다.
- 함수화 할 수 있는 부분을 함수화 했습니다. ( darkModeListen() 함수 ) 
- 주석을 추가했습니다.

ivybae replied:
@shim506 모두 확인했습니다~ resolve 하겠습니다. `// 다크 모드 동작 컨트롤` 주석은 이미 메서드명으로 동작이 짐작가능하니 필요없는 주석으로 보이네요~

ivybae commented on app/src/main/java/com/codesquad/kotlinphotoframe/MainActivity.kt:
```diff
@@ -1,11 +1,118 @@
         setContentView(R.layout.activity_main)
+
+
+        var constLaytout = findViewById<View>(R.id.const_layout)
```
@shim506 mutable, immutable 타입의 구분을 신경써주세요. View에 대한 참조를 얻어온 이후에 변경을 허용해야 할까요?

shim506 replied:
변경을 허용하지 않는 것이 좋겠습니다 감사합니다

ivybae commented on app/src/main/java/com/codesquad/kotlinphotoframe/SecondActivity.kt:
```diff
@@ -0,0 +1,58 @@
+
+        Log.d(TAG, "onCreate")
+
+        val btn_close = findViewById<Button>(R.id.btn_close)
```
@shim506 변수명은 camelCase를 따라야 합니다. 수정해주세요~

shim506 replied:
수정했습니다. 감사합니다.

ivybae commented on app/src/main/res/values/styles.xml:
```diff
@@ -0,0 +1,12 @@
+    <style name="myText" parent="TextAppearance.MaterialComponents.Subtitle2">
+
+        <item name="android:textColor">
+            @color/cardview_dark_background
```
@shim506 style의 item은 한줄로 작성합니다. 수정해주세요~

shim506 replied:
세세한 부분까지 꼼꼼히 봐주시고 코멘트 남겨주셔서 감사합니다 ㅜㅜ 수정했습니다!

ivybae reviewed (commented):
수고하셨습니다! 
style 구성 및 다크모드 지원 구현해보신 점 좋습니다! 
deprecated된 API에 대해서 정리해보신 것도 좋구요, 다만 Activity에 대해 학습한 내용이 없는 점은 아쉽습니다~ 
코멘트 내용 확인해주세요 :)

shim506 committed:
var 변수 타입 val 로 수정

<코드 정리>
- 필요하지 않은 빈줄을 정리했습니다.
- 코틀린 문법을 이용해서 if 문을 정리했습니다.
- 함수화 할 수 있는 부분을 함수화 했습니다. ( darkModeListen() 함수 )
- 주석을 추가했습니다.

shim506 committed:
- style 부분 수정

shim506 committed:
showSnackForActivityResult 함수 추가

ivybae commented on app/src/main/java/com/codesquad/kotlinphotoframe/MainActivity.kt:
```diff
@@ -43,19 +42,18 @@ class MainActivity : AppCompatActivity() {
+        showSnackForActivityResult(myButton, SecondActivity::class.java, "사진을 불러옵니다" , constLayout)
+    }
+
+    private fun showSnackForActivityResult(myButton: TextView, targetActivity: Class<SecondActivity> , msg : String , showingView: View ) {
```
@shim506 targetActivity는 메서드 인자로 받지 않아도 되지 않을까요? 전역에서 사용할 수 있는 Activity 라서요. 이 부분은 어떻게 생각하세요?

shim506 replied:
해당 함수를 SecondActivity 뿐 아닌 다른 엑티비티로도 이동할 수 있게하고 싶어서 인자로 넣었는데 인자 타입이 잘못 되어 있었네요 수정했습니다

ivybae reviewed (commented):
추가 커멘트 확인해주세요~

ivybae commented:
아직 남은 미션이 있으시니 우선 머지하겠습니다~ 커멘트 내용 확인해주세요~

shim506 commented:
> 아직 남은 미션이 있으시니 우선 머지하겠습니다~ 커멘트 내용 확인해주세요~
커멘트 확이했습니다.
머지해 주신대로 진행하겠습니다. 감사합니다
